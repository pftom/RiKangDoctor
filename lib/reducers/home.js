Object.defineProperty(exports,"__esModule",{value:true});var _immutable=require('immutable');var _immutable2=_interopRequireDefault(_immutable);


var _constants=require('../constants/');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}























var initialHomeValue=_immutable2.default.Map({
doctors:null,
posts:null,
doctor:null,
post:null,
isLoadingData:false,
loadingError:false,
loadingSuccess:false});


var home=function home(){var state=arguments.length>0&&arguments[0]!==undefined?arguments[0]:initialHomeValue;var action=arguments[1];
switch(action.type){
case _constants.GET_DOCTORS:
case _constants.GET_POSTS:
case _constants.GET_SINGLE_POST:
case _constants.GET_SINGLE_DOCTOR:

return state.merge({
isLoadingData:true,
loadingError:false,
loadingSuccess:false});


case _constants.GET_DOCTORS_SUCCESS:var

doctors=action.doctors;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
doctors:doctors});


case _constants.GET_POSTS_SUCCESS:var

posts=action.posts;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
posts:posts});



case _constants.GET_SINGLE_DOCTOR_SUCCESS:var

doctor=action.doctor;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
doctor:doctor});


case _constants.GET_SINGLE_POST_SUCCESS:var

post=action.post;
return state.merge({
isLoadingData:false,
loadingSuccess:true,
post:post});



case _constants.GET_DOCTORS_ERROR:
case _constants.GET_POSTS_ERROR:
case _constants.GET_SINGLE_DOCTOR_ERROR:
case _constants.GET_SINGLE_POST_ERROR:

return state.merge({
isLoadingData:false,
loadingError:true});


default:
return state;}

};exports.default=

home;