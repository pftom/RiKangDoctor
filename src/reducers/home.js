import Immutable from 'immutable';

//import action constants
import { 
  GET_DOCTORS,
  GET_DOCTORS_ERROR,
  GET_DOCTORS_SUCCESS, 

  GET_POSTS,
  GET_POSTS_SUCCESS,
  GET_POSTS_ERROR,
} from '../constants/';

//import single action constants
import {
  GET_SINGLE_DOCTOR,
  GET_SINGLE_DOCTOR_SUCCESS,
  GET_SINGLE_DOCTOR_ERROR,

  GET_SINGLE_POST,
  GET_SINGLE_POST_SUCCESS,
  GET_SINGLE_POST_ERROR,
} from '../constants/';



//home reducers
const initialHomeValue = Immutable.Map({
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});

const home = (state = initialHomeValue, action) => {
  switch (action.type) {
    case GET_DOCTORS:
    case GET_POSTS:
    case GET_SINGLE_POST:
    case GET_SINGLE_DOCTOR:

      return state.merge({
        isLoadingData: true,
        loadingError: false,
        loadingSuccess: false,
      });
    
    case GET_DOCTORS_SUCCESS:

      const { payload } = action;
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        doctors: payload,
      });
    
    case GET_POSTS_SUCCESS:

    //as payload has already defined
    //so use action.payload
    return state.merge({
      isLoadingData: false,
      loadingSuccess: true,
      posts: action.payload,
    });


    case GET_SINGLE_DOCTOR_SUCCESS:

        const { doctor } = action;
        return state.merge({
          isLoadingData: false,
          loadingSuccess: true,
          doctor
        });
      
      case GET_SINGLE_POST_SUCCESS:

      const { post } = action;
      return state.merge({
        isLoadingData: false,
        loadingSuccess: true,
        post,
      });
      

    case GET_DOCTORS_ERROR:
    case GET_POSTS_ERROR:
    case GET_SINGLE_DOCTOR_ERROR:
    case GET_SINGLE_POST_ERROR:

      return state.merge({
        isLoadingData: false,
        loadingError: true,
      });

    default:
      return state;
  }
};

export default home;