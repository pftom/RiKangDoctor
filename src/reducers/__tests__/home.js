import Immutable from 'immutable';

import home from '../home';

//import action constants
import { 
  GET_DOCTORS,
  GET_DOCTORS_ERROR,
  GET_DOCTORS_SUCCESS, 

  GET_POSTS,
  GET_POSTS_SUCCESS,
  GET_POSTS_ERROR,
} from '../../constants/';

//import single action constants
import {
  GET_SINGLE_DOCTOR,
  GET_SINGLE_DOCTOR_SUCCESS,
  GET_SINGLE_DOCTOR_ERROR,

  GET_SINGLE_POST,
  GET_SINGLE_POST_SUCCESS,
  GET_SINGLE_POST_ERROR,
} from '../../constants/';



//get doctors
const getDoctorsAction = {
  type: GET_DOCTORS,
};

const getDoctorsSuccessAction = {
  type: GET_DOCTORS_SUCCESS,
  doctors: {},
};

const getDoctorsErrorAction = {
  type: GET_DOCTORS_ERROR,
};

//get posts
const getPostsAction = {
  type: GET_POSTS,
};

const getPostsSuccessAction = {
  type: GET_POSTS_SUCCESS,
  posts: {},
};

const getPostsErrorAction = {
  type: GET_POSTS_ERROR,
};

//get single doctor
const getSingleDoctorAction = {
  type: GET_SINGLE_DOCTOR,
};

const getSingleDoctorSuccessAction = {
  type: GET_SINGLE_DOCTOR_SUCCESS,
  doctor: {},
};

const getSingleDoctorErrorAction = {
  type: GET_SINGLE_DOCTOR_ERROR,
};

//get single post
const getSinglePostAction = {
  type: GET_SINGLE_POST,
};

const getSinglePostSuccessAction = {
  type: GET_SINGLE_POST_SUCCESS,
  post: {},
};

const getSinglePostErrorAction = {
  type: GET_SINGLE_POST_ERROR,
};


//home reducers
const initialHomeValue = Immutable.Map({
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: false,
});

test('the home reducer work as well', () => {
  //test doctors
  expect(home(initialHomeValue, getDoctorsAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getDoctorsSuccessAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getDoctorsErrorAction)).toMatchSnapshot();

  //test home
  expect(home(initialHomeValue, getPostsAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getPostsSuccessAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getPostsErrorAction)).toMatchSnapshot();

  //test home doctor
  expect(home(initialHomeValue, getSingleDoctorAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getSingleDoctorSuccessAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getSingleDoctorErrorAction)).toMatchSnapshot();

  //test home post
  expect(home(initialHomeValue, getSinglePostAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getSinglePostSuccessAction)).toMatchSnapshot();
  expect(home(initialHomeValue.merge({ isLoadingData: true }), getSinglePostErrorAction)).toMatchSnapshot();
});