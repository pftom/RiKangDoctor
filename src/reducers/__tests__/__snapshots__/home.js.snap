// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the home reducer work as well 1`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the home reducer work as well 2`] = `
Immutable.Map {
  doctors: Immutable.Map {
  },
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the home reducer work as well 3`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the home reducer work as well 4`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the home reducer work as well 5`] = `
Immutable.Map {
  doctors: null,
  posts: Immutable.Map {
  },
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the home reducer work as well 6`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the home reducer work as well 7`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the home reducer work as well 8`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: Immutable.Map {
  },
  post: null,
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the home reducer work as well 9`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;

exports[`the home reducer work as well 10`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: true,
  loadingError: false,
  loadingSuccess: false,
}
`;

exports[`the home reducer work as well 11`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: Immutable.Map {
  },
  isLoadingData: false,
  loadingError: false,
  loadingSuccess: true,
}
`;

exports[`the home reducer work as well 12`] = `
Immutable.Map {
  doctors: null,
  posts: null,
  doctor: null,
  post: null,
  isLoadingData: false,
  loadingError: true,
  loadingSuccess: false,
}
`;
